### PR TITLE
WT-1978 Better checking and tests for index cursor comparison

### PR DESCRIPTION
**Pull request by ddanderson**

### CODE REVIEW

**Comment by ddanderson** [comment]
@michaelcahill or @agorrod, this is ready for review.

File: src/cursor/cur_index.c [hunk 1]
```diff
@@ -76,7 +76,8 @@ __curindex_compare(WT_CURSOR *a, WT_CURSOR *b, int *cmpp)
 
 	/* Check both cursors are "index:" type. */
 	if (!WT_PREFIX_MATCH(a->uri, "index:") ||
-	    !WT_PREFIX_MATCH(b->uri, "index:"))
+	    !WT_PREFIX_MATCH(b->uri, "index:") ||
```
Comment (agorrod) [inline src/cursor/cur_index.c]: @ddanderson Is this second check for ```index:``` redundant? Won't the ```strcmp``` catch it?

**Comment by agorrod** [comment]
Thanks @ddanderson nice catch. The change looks correct to me, but I think the check is more complex than necessary.

File: src/cursor/cur_index.c (hunk 1 above)
Comment (ddanderson) [inline src/cursor/cur_index.c]: @agorrod, you're right.  Thanks, I'll fix.

**Pull request merged**